### PR TITLE
Sample functions deployer

### PR DESCRIPTION
**Pull request by hiranya911**
A simple script to generate the HTTP entrypoints from a given set of function triggers.

```
$ go run deploy_functions.go github.com/inlined/go-functions/sample/lib

package main

import (
        alias "github.com/inlined/go-functions/sample/lib"
        "github.com/inlined/go-functions/support/emulator"
)

func main() {
        emulator.Serve(map[string]interface{}{
                "PubSubListener": alias.PubSubListener,
                "Webhook": alias.Webhook,
        })
}
```

### CODE REVIEW

File: deployer/deploy_functions.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	}
+	if packages.PrintErrors(pkgs) > 0 {
+		return nil, fmt.Errorf("one or mor errors")
+	}
+
+	var results []types.Object
+	scope := pkgs[0].Types.Scope()
+	for _, symbol := range scope.Names() {
+		typeInfo := scope.Lookup(symbol)
+		if isTrigger(symbol, typeInfo) {
```
Comment (inlined) [inline deployer/deploy_functions.go]: While we clearly need to make sure we don't add an unexported type, we don't really need to do the typecheck against pubsub/http functions. emulator.Serve will already ignore unsupported types and using a filter like this means that we would have to remember to update the codegen code every time we add a new trigger.

File: deployer/deploy_functions.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
+package main
```
Comment (inlined) [inline deployer/deploy_functions.go]: Since this isn't deploying any functtions, wdyt about `/support/codegen`?

**Review by inlined — APPROVED** [review]

File: deployer/deploy_functions.go (hunk 2 above)
Comment (hiranya911) [reply deployer/deploy_functions.go]: Moved to support/codegen

File: deployer/deploy_functions.go (hunk 1 above)
Comment (hiranya911) [reply deployer/deploy_functions.go]: Done. This makes code even more simple.

**Pull request merged**